### PR TITLE
test: Added tests of health checks.

### PR DESCRIPTION
**Pull request by energinet-ajw**
The test checks if all 11 health checks start up when the application starts.
Some of the health check have been renamed to be consistent.

A bit off topic:
An other added test checks that the three repeating triggers have been registered in the IoC.

## References

https://app.zenhub.com/workspaces/mandalorian-5fd22a1a59e4740018f5ab5a/issues/gh/energinet-datahub/opengeh-wholesale/1366

### CODE REVIEW

File: source/dotnet/wholesale-api/WebApi.IntegrationTests/WebApi/HealthCheckTests.cs [hunk 1]
```diff
@@ -13,50 +13,79 @@
...
-            // Act
-            var actualResponse = await Client.GetAsync(HealthChecksConstants.ReadyHealthCheckEndpointRoute);
-
-            // Assert
-            actualResponse.StatusCode.Should().Be(HttpStatusCode.OK);
-
-            var actualContent = await actualResponse.Content.ReadAsStringAsync();
-            actualContent.Should().StartWith("{\"status\":\"Healthy\"");
-        }
+        // Arrange + Act
```
Comment (BjarkeMeier) [inline source/dotnet/wholesale-api/WebApi.IntegrationTests/WebApi/HealthCheckTests.cs:40]:
```suggestion
        // Act
```

File: source/dotnet/wholesale-api/WebApi.UnitTests/RepeatingTriggerHealthChecksTests.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+using Energinet.DataHub.Wholesale.Batches.Application.Workers;
+using Energinet.DataHub.Wholesale.Events.Application.Options;
+using Energinet.DataHub.Wholesale.Events.Application.Triggers;
+using Energinet.DataHub.Wholesale.WebApi.Configuration.Modules;
+using Microsoft.Extensions.DependencyInjection;
+using Xunit;
+
+namespace Energinet.DataHub.Wholesale.WebApi.UnitTests;
+
+public class RepeatingTriggerHealthChecksTests
```
Comment (BjarkeMeier) [inline source/dotnet/wholesale-api/WebApi.UnitTests/RepeatingTriggerHealthChecksTests.cs:25]: How do these tests relate to health checks?

File: source/dotnet/wholesale-api/WebApi/HealthChecks/HealthCheck.cs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace Energinet.DataHub.Wholesale.WebApi.HealthChecks;
+
+public static class HealthCheck
```
Comment (BjarkeMeier) [inline source/dotnet/wholesale-api/WebApi/HealthChecks/HealthCheck.cs:17]:
```suggestion
public static class HealthCheckNames
```

File: source/dotnet/wholesale-api/WebApi.IntegrationTests/WebApi/HealthCheckTests.cs [hunk 4]
```diff
@@ -13,50 +13,79 @@
...
+        // Act
+        var actualResponse = await Client.GetAsync(HealthChecksConstants.ReadyHealthCheckEndpointRoute);
+
+        // Assert
+        var actualContent = await actualResponse.Content.ReadAsStringAsync();
+        expectedHealthCheckNames.ToList().ForEach(x => actualContent.Should().Contain(x));
+    }
+
+    [Fact]
+    public async Task When_RequestReadyStatus_Then_UnknownHealthCheckMustNotBeInResponse()
```
Comment (BjarkeMeier) [inline source/dotnet/wholesale-api/WebApi.IntegrationTests/WebApi/HealthCheckTests.cs:79]: Anything could go into the response. Why specifically testing for this - seamingly - arbitrary string? I think I would leave out this test and stick with your `When_RequestReadinessStatus_Then_ResponseIsOkAndHealthy` test.

File: source/dotnet/wholesale-api/WebApi.IntegrationTests/WebApi/HealthCheckTests.cs [hunk 5]
```diff
@@ -13,50 +13,79 @@
...
+        var actualContent = await actualResponse.Content.ReadAsStringAsync();
+        actualContent.Should().StartWith("{\"status\":\"Healthy\"");
+    }
+
+    [Fact]
+    public async Task When_RequestReadyStatus_Then_AllHealthChecksMustBeInResponse()
+    {
+        // Arrange
+        var type = typeof(HealthCheck);
+        var expectedHealthCheckNames = type.GetProperties(BindingFlags.Public | BindingFlags.Instance).Select(x => x.Name);
```
Comment (BjarkeMeier) [inline source/dotnet/wholesale-api/WebApi.IntegrationTests/WebApi/HealthCheckTests.cs:68]: Does `BidingFlags.Instance` relate to const fields in a static class?

File: source/dotnet/wholesale-api/WebApi.IntegrationTests/WebApi/HealthCheckTests.cs [hunk 6]
```diff
@@ -13,50 +13,79 @@
...
+
+        // Act
+        var actualResponse = await Client.GetAsync(HealthChecksConstants.ReadyHealthCheckEndpointRoute);
+
+        // Assert
+        var actualContent = await actualResponse.Content.ReadAsStringAsync();
+        expectedHealthCheckNames.ToList().ForEach(x => actualContent.Should().Contain(x));
```
Comment (BjarkeMeier) [inline source/dotnet/wholesale-api/WebApi.IntegrationTests/WebApi/HealthCheckTests.cs:75]: I think FluentAssertions has collection assertions that does not need neither `ToList()` nor `ForEach()`.

File: source/dotnet/wholesale-api/WebApi.UnitTests/RepeatingTriggerHealthChecksTests.cs (hunk 2 above)
Comment (energinet-ajw) [reply source/dotnet/wholesale-api/WebApi.UnitTests/RepeatingTriggerHealthChecksTests.cs:25]: Mistake. Renamed.

File: source/dotnet/wholesale-api/WebApi/HealthChecks/HealthCheck.cs (hunk 3 above)
Comment (energinet-ajw) [reply source/dotnet/wholesale-api/WebApi/HealthChecks/HealthCheck.cs]: Renamed.

File: source/dotnet/wholesale-api/WebApi.IntegrationTests/WebApi/HealthCheckTests.cs (hunk 6 above)
Comment (energinet-ajw) [reply source/dotnet/wholesale-api/WebApi.IntegrationTests/WebApi/HealthCheckTests.cs:75]: Looked for it but couldn't find it.

Comment (BjarkeMeier) [reply source/dotnet/wholesale-api/WebApi.IntegrationTests/WebApi/HealthCheckTests.cs:75]: My bad. I overlooked the `Contains()` part.

File: source/dotnet/wholesale-api/WebApi.UnitTests/RepeatingTriggerRegistrationTests.cs [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+namespace Energinet.DataHub.Wholesale.WebApi.UnitTests;
+
+public class RepeatingTriggerRegistrationTests
+{
+    [Theory]
+    [InlineAutoData(typeof(RegisterCompletedBatchesTrigger))]
+    [InlineAutoData(typeof(StartCalculationTrigger))]
+    [InlineAutoData(typeof(UpdateBatchExecutionStateTrigger))]
+    public void Repeating_Trigger_Must_Be_Registered_In_Ioc(Type type, ServiceBusOptions options, ServiceCollection serviceCollection)
```
Comment (BjarkeMeier) [inline source/dotnet/wholesale-api/WebApi.UnitTests/RepeatingTriggerRegistrationTests.cs:31]: nit: 
```suggestion
    public void Repeating_Trigger_Is_Registered_In_Ioc(Type type, ServiceBusOptions options, ServiceCollection serviceCollection)
```

**Review by johevemi — APPROVED** [review]

**Pull request merged**